Fitbit.Portable/FitbitClient.cs

```diff
@@ -46,18 +80,44 @@ public class FitbitClient : IFitbitClient
         /// <param name="credentials">Obtain this information from your developer dashboard. App credentials are required to perform token refresh</param>
         /// <param name="accessToken">Authenticate with Fitbit API using OAuth2. Authenticator2 class is a helper for this process</param>
         /// <param name="interceptor">An interface that enables sniffing all outgoing and incoming http requests from FitbitClient</param>
-        public FitbitClient(FitbitAppCredentials credentials, OAuth2AccessToken accessToken, IFitbitInterceptor interceptor = null, ITokenManager tokenManager = null)
+        public FitbitClient(FitbitAppCredentials credentials, OAuth2AccessToken accessToken, List <IFitbitInterceptor> interceptors, bool enableOAuth2TokenRefresh = true, ITokenManager tokenManager = null)
         {
             this.AppCredentials = credentials;
             this.AccessToken = accessToken;
-            this.MessageInterceptor = interceptor;
-            this.EnableOAuth2TokenRefresh = true;
+
+            this.FitbitInterceptorPipeline = new List<IFitbitInterceptor>();
+
+            if(interceptors != null && interceptors.Count > 0)
+                this.FitbitInterceptorPipeline.AddRange(interceptors);
 
             ConfigureTokenManager(tokenManager);
 
+            //Auto refresh should always be the last handle to be registered.
+            ConfigureAutoRefresh(enableOAuth2TokenRefresh);
             CreateHttpClientForOAuth2();
         }
 
+
+        public FitbitClient(FitbitAppCredentials credentials, OAuth2AccessToken accessToken, bool enableOAuth2TokenRefresh) : this(credentials, accessToken, null, enableOAuth2TokenRefresh)
```

Comment:
Don't think all these constructor overrides are needed (thought I'd raised this but can't find the note anywhere ... ).